File: dotnet/test/Azure.Iot.Operations.Protocol.MetlTests/DefaultPrologue.cs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

namespace Azure.Iot.Operations.Protocol.MetlTests
{
    public class DefaultPrologue
    {
        public DefaultPrologue()
        {
            Executor = new();
            Invoker = new();
            Receiver = new();
            Sender = new();
        }

        public DefaultExecutor Executor { get; set; }

        public DefaultInvoker Invoker { get; set; }

        public DefaultReceiver Receiver { get; set; }

        public DefaultSender Sender { get; set; }
    }
}

File: rust/azure_iot_operations_protocol/tests/metl/default_prologue.rs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

use serde::Deserialize;

use crate::metl::default_executor::DefaultExecutor;
use crate::metl::default_invoker::DefaultInvoker;
use crate::metl::default_receiver::DefaultReceiver;
use crate::metl::default_sender::DefaultSender;

#[derive(Deserialize, Debug)]
pub struct DefaultPrologue {
    #[serde(rename = "executor")]
    pub executor: Option<DefaultExecutor>,

    #[serde(rename = "invoker")]
    pub invoker: Option<DefaultInvoker>,

    #[serde(rename = "receiver")]
    pub receiver: Option<DefaultReceiver>,

    #[serde(rename = "sender")]
    pub sender: Option<DefaultSender>,
}
